src/models.py

```diff
@@ -136,13 +140,42 @@ def callbacks_unet(callbacks_config):
                    ])
 
 
-def multiclass_weighted_cross_entropy(output, target, w0, sigma, C):
+def multiclass_weighted_cross_entropy(output, target, weights_function=None):
+    """Calculate weighted Cross Entropy loss for multiple classes.
+
+    This function calculates torch.nn.CrossEntropyLoss(), but each pixel loss is weighted.
+    Target for weights is defined as a part of target, in target[:, 1:, :, :].
+    If weights_function is not None weights are calculated by applying this function on target[:, 1:, :, :].
+    If weights_function is None weights are taken from target[:, 1, :, :].
+
+    Args:
+        output (torch.Tensor): Model output of shape (N x C x H x W).
+        target (torch.Tensor): Target of shape (N x (1 + K) x H x W). Where K is number of different weights.
+        weights_function (function, optional): Function applied on target for weights.
+
+    Returns:
+        float: Loss value.
```

Comment:
@taraspiotr are you sure pytorch 0.3X version has this as a float and not tensor?

Reply:
@jakubczakon you are right.